overlord/devicestate/devicestate.go

```diff
@@ -1170,13 +1178,25 @@ func checkForRequiredSnapsNotInModel(model *asserts.Model, vSets *snapasserts.Va
 	return nil
 }
 
+func checkModelSnapsCanBePresentInValidationSets(model *asserts.Model, vSets *snapasserts.ValidationSets) error {
```

Comment:
is the a concrete difference between this and checkForInvalidSnapsInModel below?

Reply:
You're right, I merged the two.